README.md

```diff
@@ -166,543 +119,10 @@ There are two parameters:
-	// same code as in DecodeMsgpack
-	return nil
-}
-
-func encodeTuple(e *msgpack.Encoder, v reflect.Value) error {
-	c := v.Interface().(Tuple)
-	// same code as in EncodeMsgpack
-	return nil
-}
-
-func decodeTuple(d *msgpack.Decoder, v reflect.Value) error {
-	c := v.Addr().Interface().(*Tuple)
-	// same code as in DecodeMsgpack
-	return nil
-}
-*/
-
-```
-
-## Options
-
-* `Timeout` - timeout for any particular request. If `Timeout` is zero request,
-  any request may block infinitely.
-* `Reconnect` - timeout between reconnect attempts. If `Reconnect` is zero, no
-  reconnects will be performed.
-* `MaxReconnects` - maximal number of reconnect failures; after that we give it
-  up. If `MaxReconnects` is zero, the client will try to reconnect endlessly.
-* `User` - user name to log into Tarantool.
-* `Pass` - user password to log into Tarantool.
-
-## Working with queue
-```go
-package main
-import (
-	"gopkg.in/vmihailenco/msgpack.v2"
-	"github.com/tarantool/go-tarantool"
-	"github.com/tarantool/go-tarantool/queue"
-	"time"
-	"fmt"
-	"log"
-)
-
-type customData struct{
-	Dummy bool
-}
-
-func (c *customData) DecodeMsgpack(d *msgpack.Decoder) error {
-	var err error
-	if c.Dummy, err = d.DecodeBool(); err != nil {
-		return err
-	}
-	return nil
-}
-
-func (c *customData) EncodeMsgpack(e *msgpack.Encoder) error {
-	return e.EncodeBool(c.Dummy)
-}
-
-func main() {
-	opts := tarantool.Opts{
-		Timeout: time.Second,
-		Reconnect: time.Second,
-		MaxReconnects: 5,
-		User: "user",
-		Pass: "pass",
-		// ...
-	}
-	conn, err := tarantool.Connect("127.0.0.1:3301", opts)
-
-	if err != nil {
-		log.Fatalf("connection: %s", err)
-		return
-	}
-
-	cfg := queue.Cfg{
-		Temporary:  true,
-		IfNotExists: true,
-		Kind:       queue.FIFO,
-		Opts: queue.Opts{
-			Ttl:   10 * time.Second,
-			Ttr:   5 * time.Second,
-			Delay: 3 * time.Second,
-			Pri:   1,
-		},
-	}
-
-	que := queue.New(conn, "test_queue")
-	if err = que.Create(cfg); err != nil {
-		log.Fatalf("queue create: %s", err)
-		return
-	}
-
-	// put data
-	task, err := que.Put("test_data")
-	if err != nil {
-		log.Fatalf("put task: %s", err)
-	}
-	fmt.Println("Task id is", task.Id())
-
-	// take data
-	task, err = que.Take() //blocking operation
-	if err != nil {
-		log.Fatalf("take task: %s", err)
-	}
-	fmt.Println("Data is", task.Data())
-	task.Ack()
-
-	// take typed example
-	putData := customData{}
-	// put data
-	task, err = que.Put(&putData)
-	if err != nil {
-		log.Fatalf("put typed task: %s", err)
-	}
-	fmt.Println("Task id is ", task.Id())
-
-	takeData := customData{}
-	//take data
-	task, err = que.TakeTyped(&takeData) //blocking operation
-	if err != nil {
-		log.Fatalf("take take typed: %s", err)
-	}
-	fmt.Println("Data is ", takeData)
-	// same data
-	fmt.Println("Data is ", task.Data())
-
-	task, err = que.Put([]int{1, 2, 3})
-	task.Bury()
-
-	task, err = que.TakeTimeout(2 * time.Second)
-	if task == nil {
-		fmt.Println("Task is nil")
-	}
-
-	que.Drop()
-}
-```
```

Comment:
Features of implementation:

- If you use the connection timeout and call `TakeWithTimeout` with a parameter greater than the connection timeout, the parameter is reduced to it.
- If you use the connection timeout and call `Take`, we return an error if we cannot take the task out of the queue within the time corresponding to the connection timeout.

Reply:
Updated.